src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -406,17 +406,39 @@ private SyntaxNode GetNodeContainingTargetNode(SyntaxNode matchingNode)
             var reorderedParameters = updatedSignature.UpdatedConfiguration.ToListOfParameters();
 
             var newParameters = new List<T>();
+            int index = 0;
             foreach (var newParam in reorderedParameters)
             {
                 var pos = originalParameters.IndexOf(newParam);
                 var param = list[pos];
```

Comment:
so I guess expectation is new param alwasy exist in old parameter list? so add new parameter won't hit this code path? is adding new parameter doesn't have this issue?

Reply:
Correct. AddParameter doesn't seem to use ChangeSignatureService.